bgpd/bgp_route.c

```diff
@@ -8409,14 +8409,31 @@ static int bgp_show_table(struct vty *vty, struct bgp *bgp, safi_t safi,
 				continue;
 
 			p = &rn->p;
-			sprintf(buf2, "%s/%d",
-				inet_ntop(p->family, &p->u.prefix, buf, BUFSIZ),
-				p->prefixlen);
-			if (first)
-				vty_out(vty, "\"%s\": ", buf2);
-			else
-				vty_out(vty, ",\"%s\": ", buf2);
-
+			/* encode prefix */
```

Comment:
Shouldn't this just be in prefix2str and we get rid of all this special case?

Reply:
deciphering flowspec requires some code to be move from bgp to lib.
if possible, I would propose prefix api to have a registration API for some extended family like flowspec.

Reply:
I guess I don't understand why a callback would be necessary.  Can you help me to understand?  What about a flowspec prefix cannot be abstracted into `struct prefix` as part of the display.  Why would we not want that displaying to be consistent across all daemons that may want to use it.

Reply:
Flowspec has no meaning out of BGP.
However, BGP Flowspec relies on the definition of prefixes to store its information as opaque.
I used prefix library to host bgp flowspec as an opaque value ( length in bytes + pointer).

I don't want to move the decoding/encoding information from BGP to LIB, because noone except BGP will benefit from it.

any opinion?

Reply:
I don't see flowspec being carried any way other than BGP in the future -- so I tend to think it's okay where it is in the fix for right now.